package_test.go

```diff
@@ -194,6 +197,20 @@ func (s *PackageSuite) TestValidIterGet(c *C) {
 		inputs:   []any{Address{ID: 1000}},
 		outputs:  [][]any{},
 		expected: [][]any{},
+	}, {
+		summary:  "simple in",
+		query:    "SELECT * AS &Person.* FROM person WHERE id IN ($S[:])",
+		types:    []any{Person{}, sqlair.S{}},
+		inputs:   []any{sqlair.S{30, 35, 36, 37, 38, 39, 40}},
+		outputs:  [][]any{{&Person{}}, {&Person{}}, {&Person{}}},
+		expected: [][]any{{&Person{30, "Fred", 1000}}, {&Person{40, "Mary", 3500}}, {&Person{35, "James", 4500}}},
+	}, {
+		summary:  "complex in",
+		query:    "SELECT * AS &Person.* FROM person WHERE id IN ($Person.id, $S[:], $Manager.id, $IntSlice[:], $StringSlice[:])",
```

Comment:
This is only valid syntax by virtue of us not parsing the IN statement. All the input expressions in this statement are parsed separately independent of each other (i.e. the comma between each one is a `bypass` part). To reject this syntax we would have to actually parse IN statements and be far more context aware than we are with parsing.

I agree that there is no need to encouraged it however.  In general, if there is a valid use case like this that we do not recommend, should it be included in the tests?

Reply:
I think we should catch it early. If we detect slice syntax, there should be only member in the `IN` list.

Reply:
I have some reservations about doing this. In the rest of SQLair we avoid parsing any query syntax other than that which is strictly necessary. For example with output expressions we do not even check that they occur after a `SELECT`. This was a conscious decision to avoid as much dependency on the SQL syntax itself as possible and allow us to handle unforeseen usecases and syntax in other dialects. 

When initially discussing this feature with Gustavo it was decided that we would not tie the slice support to `IN` statements and instead allow them to be used anywhere in the query so as not to restrict support to only the use case we could foresee.

I am also not sure that I understand why allowing other values in an `IN` statement beside the slice is a problem, though I do agree that it is bad practice.  There might be cases such as `... WHERE x IN ($S[:], "invalid")` where the slice from the user only captures a subset of values that should be matched against.

Reply:
When you say that they are placeholders in the same clause are you referring to the parentheses around the input expressions? 

In the implementation as it is SQLair doesn't know that these are in the same clause.
It parses:
`IN($S[:], $T[:])`
as:
`Bypass["IN("], Input["S[:]"], Bypass[", "], Input["T[:]"], Bypass[")"]`
The bypass parts could be anything as far as SQLair is concerned, including the end of the clause. When we generate the SQL to replace the slice input expression we do not generate any parentheses.

Reply:
Just to add to Alastair comment, when encountering a slice all we generate in the final SQL is
```
@sqlair_0, @sqlair_1, ..., @sqlair_n
```
we do not include parentheses.